crates/polars-ops/src/chunked_array/strings/escape_regex.rs

```diff
@@ -0,0 +1,16 @@
+use polars_core::prelude::arity::unary_elementwise;
+use polars_core::prelude::StringChunked;
+use regex::escape;
+
+#[inline]
+pub fn escape_regex_str(s: &str) -> String {
+    escape(s)
+}
+
+fn escape_regex_helper(s: Option<&str>) -> Option<String> {
+    s.map(escape_regex_str)
+}
+
+pub fn escape_regex(ca: &StringChunked) -> StringChunked {
+    unary_elementwise(ca, escape_regex_helper)
```

Comment:
Can you add a dependency on `regex_syntax` (we already have this indirectly), and use a loop similar to this?

```rust
pub fn escape_regex(ca: &StringChunked) -> StringChunked {
    let mut buffer = String::new();
    let mut builder = StringChunkedBuilder::new(ca.name().clone(), ca.len());
    for opt_s in ca.iter() {
        if let Some(s) = opt_s {
            buffer.clear();
            regex_syntax::escape_into(s, &mut buffer);
            builder.append_value(&buffer);
        } else {
            builder.append_null();
        }
    }
}
```

This prevents us from having to allocate for each string.